definitions/artifacts/gcp-service-account.json

```diff
@@ -45,19 +45,19 @@
         "auth_provider_x509_cert_url": {
           "type": "string",
           "title": "Auth Provider x509 Certificate URL",
-          "description": "",
+          "description": "Auth Provider x509 Certificate URL",
           "default": "https://www.googleapis.com/oauth2/v1/certs"
         },
         "auth_uri": {
-          "type": "string",
           "title": "Auth URI",
-          "description": "",
-          "default": "https://accounts.google.com/o/oauth2/auth"
+          "description": "Auth URI",
+          "default": "https://accounts.google.com/o/oauth2/auth",
+          "$ref": "../types/https-url.json"
```

Comment:
Should we use this type for _all_ urls, so we get the regex "for free"? Personally I like leaning on well-scoped types that have good defaults for title, description, regex, etc... and then overriding some of those fields in the artifacts.

Reply:
In the event of a `$ref`, any existing fields which collide will be retained.

So if you give this a `title` of "Auth URI" and the `https.json` ref also has a generic title "An HTTPS URL", your title will be kept.  So you end up getting a union of both, with your own fields taking precedence.

In this case, if you need a custom regex, I suggest you make your own pattern. But if the generic validation is sufficient, this is an easy way to use it.

Reply:
I didn't see it :)

And I like this pattern for our generics a lot!